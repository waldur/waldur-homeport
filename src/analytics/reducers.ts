import { Action } from '@waldur/core/reducerActions';

import * as constants from './constants';
import { Payload, State } from './types';

const INITIAL_STATE: State = {
  loading: false,
  errors: [],
  projects: [],
  tenants: [],
  searchValue: '',
  quotasHistory: {},
};

export const reducer = (
  state: State = INITIAL_STATE,
  action: Action<Payload>,
) => {
  const { type, payload } = action;
  switch (type) {
    case constants.ANALYTICS_PROJECTS_FETCH:
      return {
        ...state,
        loading: true,
      };
    case constants.ANALYTICS_PROJECTS_FETCH_SUCCESS:
      return {
        ...state,
        projects: payload.projects,
        loading: false,
      };
    case constants.ANALYTICS_PROJECTS_FETCH_ERROR:
      return {
        ...state,
        errors: [...state.errors, payload.error],
        loading: false,
      };
    case constants.ANALYTICS_TENANTS_FETCH_SUCCESS:
      return {
        ...state,
        tenants: payload.tenants,
      };
    case constants.ANALYTICS_TENANTS_FETCH_ERROR:
      return {
        ...state,
        errors: [...state.errors, payload.error],
      };
    case constants.ANALYTICS_SEARCH_FILTER_CHANGE:
      return {
        ...state,
        searchValue: payload.searchValue,
      };
    case constants.ANALYTICS_HISTORY_QUOTA_FETCH:
      return {
        ...state,
        quotasHistory: {
          ...state.quotasHistory,
          [payload.tenantUuid]: {
            ...state.quotasHistory[payload.tenantUuid],
            [payload.quotaUuid]: {
              loading: true,
              uuid: payload.quotaUuid,
            },
          },
        },
      };
    case constants.ANALYTICS_HISTORY_QUOTA_FETCH_SUCCESS:
      return {
        ...state,
        quotasHistory: {
          ...state.quotasHistory,
          [payload.tenantUuid]: {
            ...state.quotasHistory[payload.tenantUuid],
            [payload.quotaUuid]: {
              ...state.quotasHistory[payload.tenantUuid][payload.quotaUuid],
              loading: false,
              data: payload.quotas,
            },
          },
        },
      };
    case constants.ANALYTICS_HISTORY_QUOTA_FETCH_ERROR:
      return {
        ...state,
        errors: [...state.errors, payload.error],
        quotasHistory: {
          ...state.quotasHistory,
          [payload.tenantUuid]: {
            ...state.quotasHistory[payload.tenantUuid],
            [payload.quotaUuid]: {
              ...state.quotasHistory[payload.tenantUuid][payload.quotaUuid],
              loading: false,
              erred: true,
            },
          },
        },
      };
    default:
      return state;
  }
};
