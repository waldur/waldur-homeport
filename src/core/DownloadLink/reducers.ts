import {
  DOWNLOAD_REQUEST,
  DOWNLOAD_SUCCESS,
  DOWNLOAD_FAILURE,
  DOWNLOAD_RESET,
} from './constants';

const INITIAL_STATE = {
  loading: false,
  loaded: false,
  erred: false,
};

export function reducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case DOWNLOAD_REQUEST:
      return {
        loading: true,
        loaded: false,
        erred: false,
      };

    case DOWNLOAD_SUCCESS:
      return {
        loading: false,
        loaded: true,
        erred: false,
      };

    case DOWNLOAD_FAILURE:
      return {
        loading: false,
        loaded: false,
        erred: true,
      };

    case DOWNLOAD_RESET:
      return INITIAL_STATE;

    default:
      return state;
  }
}

export const getDownloadLinkState = state => state.downloadLink;
