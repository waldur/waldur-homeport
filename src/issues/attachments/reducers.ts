import { Action } from '@waldur/core/reducerActions';

import * as constants from './constants';
import { Payload, State } from './types';

const INITIAL_STATE: State = {
  loading: false,
  errors: [],
  items: [],
  uploading: 0,
  deleting: {},
  filter: constants.ISSUE_ATTACHMENTS_FILTER_NAMES.name,
};

export const reducer = (
  state: State = INITIAL_STATE,
  action: Action<Payload>,
) => {
  const { type, payload } = action;
  switch (type) {
    case constants.ISSUE_ATTACHMENTS_GET:
      return {
        ...state,
        loading: true,
      };
    case constants.ISSUE_ATTACHMENTS_GET_SUCCESS:
      return {
        ...state,
        loading: false,
        items: payload.items,
      };
    case constants.ISSUE_ATTACHMENTS_GET_ERROR:
      return {
        ...state,
        errors: [...state.errors, payload.error],
        loading: false,
      };
    case constants.ISSUE_ATTACHMENTS_PUT_START:
      return {
        ...state,
        uploading: state.uploading + payload.uploading,
      };
    case constants.ISSUE_ATTACHMENTS_PUT_SUCCESS:
      return {
        ...state,
        items: [...state.items, payload.item],
        uploading: state.uploading > 0 ? state.uploading - 1 : 0,
      };
    case constants.ISSUE_ATTACHMENTS_PUT_ERROR:
      return {
        ...state,
        errors: [...state.errors, payload.error],
        uploading: state.uploading > 0 ? state.uploading - 1 : 0,
      };
    case constants.ISSUE_ATTACHMENTS_PUT_REJECT:
      return {
        ...state,
        uploading: state.uploading > 0 ? state.uploading - 1 : 0,
      };
    case constants.ISSUE_ATTACHMENTS_DELETE_START:
      return {
        ...state,
        deleting: {
          ...state.deleting,
          [payload.uuid]: true,
        },
      };
    case constants.ISSUE_ATTACHMENTS_DELETE_SUCCESS:
      return {
        ...state,
        items: state.items.filter(item => item.uuid !== payload.uuid),
        deleting: {
          ...state.deleting,
          [payload.uuid]: null,
        },
      };
    case constants.ISSUE_ATTACHMENTS_DELETE_ERROR:
      return {
        ...state,
        deleting: {
          ...state.deleting,
          [payload.uuid]: false,
        },
        errors: [...state.errors, payload.error],
      };
    default:
      return state;
  }
};
