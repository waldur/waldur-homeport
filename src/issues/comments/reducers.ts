import { Action } from '@waldur/core/reducerActions';
import { omit } from '@waldur/core/utils';
import {
  ISSUE_ATTACHMENTS_DELETE_SUCCESS,
  ISSUE_ATTACHMENTS_PUT_SUCCESS,
} from '@waldur/issues/attachments/constants';

import * as constants from './constants';
import { Payload, State } from './types';

const INITIAL_STATE: State = {
  loading: false,
  errors: [],
  items: [],
  deleting: {},
  activeFormId: null,
  pendingAttachments: [],
  issue: null,
  uiDisabled: false,
  getErred: false,
};

export const reducer = (
  state: State = INITIAL_STATE,
  action: Action<Payload>,
) => {
  const { type, payload } = action;
  switch (type) {
    case constants.ISSUE_COMMENTS_GET:
      return {
        ...state,
        loading: true,
      };
    case constants.ISSUE_COMMENTS_GET_SUCCESS:
      return {
        ...state,
        loading: false,
        getErred: false,
        items: payload.items,
      };
    case constants.ISSUE_COMMENTS_GET_ERROR:
      return {
        ...state,
        loading: false,
        getErred: true,
        errors: [...state.errors, payload.error],
      };
    case constants.ISSUE_COMMENTS_CREATE_SUCCESS:
      return {
        ...state,
        items: [payload.item, ...state.items],
      };
    case constants.ISSUE_COMMENTS_CREATE_ERROR:
      return {
        ...state,
        errors: [...state.errors, payload.error],
      };
    case constants.ISSUE_COMMENTS_UPDATE_SUCCESS:
      return {
        ...state,
        items: state.items.map(item =>
          item.uuid === payload.item.uuid ? payload.item : item,
        ),
      };
    case constants.ISSUE_COMMENTS_UPDATE_ERROR:
      return {
        ...state,
        errors: [...state.errors, payload.error],
      };
    case constants.ISSUE_COMMENTS_DELETE:
      return {
        ...state,
        deleting: {
          ...state.deleting,
          [payload.commentId]: true,
        },
      };
    case constants.ISSUE_COMMENTS_DELETE_SUCCESS:
      return {
        ...state,
        deleting: omit(state.deleting, payload.commentId),
        items: state.items.filter(item => item.uuid !== payload.commentId),
      };
    case constants.ISSUE_COMMENTS_DELETE_ERROR:
      return {
        ...state,
        deleting: omit(state.deleting, payload.commentId),
        errors: [...state.errors, payload.error],
      };
    case ISSUE_ATTACHMENTS_PUT_SUCCESS:
      return {
        ...state,
        pendingAttachments: [...state.pendingAttachments, payload.item],
      };
    case ISSUE_ATTACHMENTS_DELETE_SUCCESS:
      return {
        ...state,
        pendingAttachments: state.pendingAttachments.filter(
          attachment => attachment.uuid !== payload.uuid,
        ),
      };
    case constants.ISSUE_COMMENTS_PENDING_ATTACHMENTS_RESET:
      return {
        ...state,
        pendingAttachments: [],
      };
    case constants.ISSUE_COMMENTS_FORM_TOGGLE:
      return {
        ...state,
        activeFormId:
          state.activeFormId !== payload.formId ? payload.formId : null,
      };
    case constants.ISSUE_COMMENTS_UI_DISABLE:
      return {
        ...state,
        uiDisabled: payload.uiDisabled,
      };
    case constants.ISSUE_COMMENTS_ISSUE_SET:
      return {
        ...state,
        issue: payload.issue,
      };
    default:
      return state;
  }
};
