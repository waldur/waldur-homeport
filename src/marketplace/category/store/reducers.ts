import * as constants from './constants';

const INITIAL_STATE = {
  filterQuery: undefined,
  sections: [],
  loading: true,
  loaded: false,
  erred: false,
};

export const categoryReducer = (state = INITIAL_STATE, action) => {
  const { type, payload } = action;
  switch (type) {
    case constants.SET_FILTER_QUERY:
      return {
        ...state,
        filterQuery: payload.filterQuery,
      };

    case constants.LOAD_DATA_START:
      return {
        ...state,
        loading: true,
        loaded: false,
        erred: false,
      };

    case constants.LOAD_DATA_SUCCESS:
      return {
        ...state,
        loading: false,
        loaded: true,
        erred: false,
        sections: payload.sections,
      };

    case constants.LOAD_DATA_ERROR:
      return {
        ...state,
        loading: false,
        loaded: false,
        erred: true,
      };

    default:
      return state;
  }
};

export const categoryOfferingsReducer = (state = { items: [] }, action) => {
  const { type, payload } = action;
  switch (type) {
    case constants.LOAD_OFFERINGS_START:
      return {
        ...state,
        loading: true,
        loaded: false,
        erred: false,
      };

    case constants.LOAD_OFFERINGS_SUCCESS:
      return {
        ...state,
        loading: false,
        loaded: true,
        erred: false,
        items: payload.items,
      };

    case constants.LOAD_OFFERINGS_ERROR:
      return {
        ...state,
        loading: false,
        loaded: false,
        erred: true,
      };

    default:
      return state;
  }
};
