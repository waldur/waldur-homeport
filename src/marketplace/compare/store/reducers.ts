import * as constants from './constants';

const INITIAL_STATE = {
  items: [],
};

export const comparisonReducer = (state = INITIAL_STATE, action) => {
  const { type, payload } = action;
  switch (type) {
    case constants.ADD_ITEM:
      return {
        ...state,
        items: [...state.items, payload.item],
      };
    case constants.REMOVE_ITEM:
      return {
        ...state,
        items: state.items.filter((item) => item.uuid !== payload.item.uuid),
      };
    case constants.SET_ITEMS:
      return {
        ...state,
        items: payload.items,
      };
    default:
      return state;
  }
};
