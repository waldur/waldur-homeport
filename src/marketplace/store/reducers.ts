import { combineReducers } from 'redux';

import { cartReducer } from '@waldur/marketplace/cart/store/reducer';
import {
  categoryReducer,
  categoryOfferingsReducer,
} from '@waldur/marketplace/category/store/reducers';
import { comparisonReducer } from '@waldur/marketplace/compare/store/reducers';
import { landingReducer } from '@waldur/marketplace/landing/store/reducer';
import { offeringReducer } from '@waldur/marketplace/offerings/store/reducer';
import { ordersReducer } from '@waldur/marketplace/orders/store/reducer';
import { serviceProviderReducer } from '@waldur/marketplace/service-providers/store/reducer';

export const reducer = combineReducers({
  comparison: comparisonReducer,
  cart: cartReducer,
  offering: offeringReducer,
  landing: landingReducer,
  orders: ordersReducer,
  category: categoryReducer,
  categoryOfferings: categoryOfferingsReducer,
  serviceProvider: serviceProviderReducer,
});
