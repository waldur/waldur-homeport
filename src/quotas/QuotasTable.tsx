import { useMemo } from 'react';
import { Col, Row } from 'react-bootstrap';
import { useSelector } from 'react-redux';

import { EChart } from '@waldur/core/EChart';
import { ImagePlaceholder } from '@waldur/core/ImagePlaceholder';
import { translate } from '@waldur/i18n';
import { themeSelector } from '@waldur/navigation/theme/store';

import { QUOTA_CATEGORIES } from './constants';
import { Quota } from './types';
import { formatQuotaValue, formatQuotaName } from './utils';

interface Resource {
  quotas: Quota[];
}

const ResourceQuotaChart = ({ value, max }) => {
  const empty = Math.max(max - value, 0);

  let color = '#108F10';
  const percentage = (value / max) * 100;
  if (percentage > 95) color = '#CC0808';
  else if (percentage > 60) color = '#FCCF5C';

  const theme = useSelector(themeSelector);

  const chartOptions = {
    tooltip: {
      trigger: 'item',
    },
    series: [
      {
        type: 'pie',
        radius: '100%',
        center: ['50%', '50%'],
        label: { show: false },
        data: [
          { value, name: translate('Used'), itemStyle: { color } },
          {
            value: empty,
            name: translate('Free'),
            itemStyle: { color: theme === 'light' ? '#f5f8fa' : '#1a261d' },
          },
        ],
        animation: false,
        emphasis: {
          itemStyle: {
            shadowColor: 'rgba(0, 0, 0, 0.5)',
          },
        },
      },
    ],
  };

  return max === Infinity ? (
    <div className="symbol symbol-65px symbol-circle">
      <ImagePlaceholder
        width="65px"
        height="65px"
        backgroundColor="transparent"
      >
        <div className="symbol-label fs-1 fw-bolder text-muted w-100 h-100">
          {value}
        </div>
      </ImagePlaceholder>
    </div>
  ) : (
    <EChart options={chartOptions} height="65px" width="65px" />
  );
};

const ResourceQuotaItem = ({ quota }: { quota: Quota }) => {
  const data = useMemo(() => {
    const formattedUsage = formatQuotaValue(quota.usage, quota.name);
    const formattedLimit = formatQuotaValue(quota.limit, quota.name);
    let usage, usageValue;
    if (formattedUsage === '∞') {
      usage = formattedUsage;
      usageValue = Infinity;
    } else {
      usage = String(formattedUsage).match(/\d+/)[0];
      usageValue = Number(usage);
    }
    let limitValue;
    const limit = formattedLimit;
    if (formattedLimit === '∞') {
      limitValue = Infinity;
    } else {
      limitValue = Number(String(formattedLimit).match(/\d+/)[0]);
    }
    return { usage, usageValue, limit, limitValue };
  }, [quota]);

  return (
    <div className="resource-quota-item fw-bold d-flex mb-6">
      <ResourceQuotaChart value={data.usageValue} max={data.limitValue} />
      <div className="ms-3">
        <span className="d-block">
          <span className="fs-4">{data.usage}</span>
          <span className="fs-7">/{data.limit}</span>
        </span>
        <span className="d-block fs-7">{formatQuotaName(quota.name)}</span>
      </div>
    </div>
  );
};

export const QuotasTable = ({ resource }: { resource: Resource }) => {
  const quotaCategories = useMemo(() => {
    const categories = Object.entries(QUOTA_CATEGORIES)
      .map(([key, category]) => {
        const quotas = resource.quotas
          .filter((q) => {
            return category.names.some((name) =>
              name instanceof RegExp ? name.test(q.name) : name === q.name,
            );
          })
          .sort((q1, q2) => q1.name.localeCompare(q2.name));
        if (quotas.length > 0) {
          return { key, label: category.label, quotas };
        } else return null;
      })
      .filter(Boolean);

    const categorizedNames = Object.values(QUOTA_CATEGORIES)
      .map((cat) => cat.names)
      .flat();
    const additional = resource.quotas
      .filter(
        (q) =>
          !categorizedNames.some((name) =>
            name instanceof RegExp ? name.test(q.name) : q.name === name,
          ),
      )
      .sort((q1, q2) => q1.name.localeCompare(q2.name));
    return { categories, additional };
  }, [resource]);

  return resource.quotas.length === 0 ? (
    <div className="row text-center">
      {translate('You have no quotas yet.')}
    </div>
  ) : (
    <div className="provider-quotas">
      {quotaCategories.categories.map((group) => (
        <div className="my-6" key={group.key}>
          <h5 className="fw-bold mb-6">{group.label}</h5>
          <Row>
            {group.quotas.map((quota) => (
              <Col xs={6} md={4} lg={3} key={quota.name}>
                <ResourceQuotaItem quota={quota} />
              </Col>
            ))}
          </Row>
        </div>
      ))}
      {quotaCategories.additional.length > 0 && (
        <div>
          <h5 className="fw-bold mb-6">{translate('Additional quotas')}</h5>
          {quotaCategories.additional.map((quota) => (
            <Row key={quota.name} className="mb-2">
              <Col sm={4} xs={6}>
                {formatQuotaName(quota.name)}:
              </Col>
              <Col sm={8} xs={6}>
                {translate('{usage} of {limit}', {
                  usage: formatQuotaValue(quota.usage, quota.name),
                  limit: formatQuotaValue(quota.limit, quota.name),
                })}
              </Col>
            </Row>
          ))}
        </div>
      )}
    </div>
  );
};
