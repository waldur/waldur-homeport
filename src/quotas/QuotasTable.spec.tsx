import { render } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { RootState } from '@waldur/store/reducers';

import { QuotasTable } from './QuotasTable';

const quotas = require('./QuotasTable.fixture.json');

const renderComponent = (resource) => {
  const mockStore = configureStore();
  const storeData: Partial<RootState> = {
    theme: {
      themes: [],
      theme: 'dark',
    },
  };
  const store = mockStore(storeData);

  const component = (
    <Provider store={store}>
      <QuotasTable resource={resource} />
    </Provider>
  );
  const wrapper = render(component);
  return wrapper.html();
};

describe('QuotasTable', () => {
  it('renders table for quotas', () => {
    const html = renderComponent({ quotas });
    expect(html).toMatchSnapshot();
  });
});
