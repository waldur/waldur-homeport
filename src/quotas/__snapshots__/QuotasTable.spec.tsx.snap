// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotasTable renders table for quotas 1`] = `"<div class=\\"my-6\\"><h5 class=\\"fw-bold mb-6\\">Compute</h5><div class=\\"row\\"><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">5</span><span class=\\"fs-7\\">/30</span></span><span class=\\"d-block fs-7\\">Instances count</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">5</span><span class=\\"fs-7\\">/128 GB</span></span><span class=\\"d-block fs-7\\">RAM</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">3</span><span class=\\"fs-7\\">/64</span></span><span class=\\"d-block fs-7\\">vCPU count</span></div></div></div></div></div><div class=\\"my-6\\"><h5 class=\\"fw-bold mb-6\\">Network</h5><div class=\\"row\\"><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">6</span><span class=\\"fs-7\\">/50</span></span><span class=\\"d-block fs-7\\">Floating IP count</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">1</span><span class=\\"fs-7\\">/100</span></span><span class=\\"d-block fs-7\\">Network count</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">8</span><span class=\\"fs-7\\">/100</span></span><span class=\\"d-block fs-7\\">Security group count</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">14</span><span class=\\"fs-7\\">/100</span></span><span class=\\"d-block fs-7\\">Security group rule count</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">1</span><span class=\\"fs-7\\">/100</span></span><span class=\\"d-block fs-7\\">Subnet count</span></div></div></div></div></div><div class=\\"my-6\\"><h5 class=\\"fw-bold mb-6\\">Storage</h5><div class=\\"row\\"><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">2</span><span class=\\"fs-7\\">/50</span></span><span class=\\"d-block fs-7\\">Snapshots count</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"symbol symbol-65px symbol-circle\\"><div class=\\"image-placeholder\\" style=\\"width:65px;min-width:65px;height:65px;background-color:transparent\\"><div class=\\"symbol-label fs-1 fw-bolder text-muted w-100 h-100\\">50</div></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">50</span><span class=\\"fs-7\\">/&#x221E;</span></span><span class=\\"d-block fs-7\\">Snapshots size</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">295</span><span class=\\"fs-7\\">/295 GB</span></span><span class=\\"d-block fs-7\\">Storage</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"content-center-center\\" style=\\"width:65px;height:65px\\"><div class style=\\"width:65px;height:65px\\"></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">8</span><span class=\\"fs-7\\">/50</span></span><span class=\\"d-block fs-7\\">Volumes count</span></div></div></div><div class=\\"col-lg-3 col-md-4 col-6\\"><div class=\\"resource-quota-item fw-bold d-flex mb-6\\"><div class=\\"symbol symbol-65px symbol-circle\\"><div class=\\"image-placeholder\\" style=\\"width:65px;min-width:65px;height:65px;background-color:transparent\\"><div class=\\"symbol-label fs-1 fw-bolder text-muted w-100 h-100\\">245</div></div></div><div class=\\"ms-3\\"><span class=\\"d-block\\"><span class=\\"fs-4\\">245</span><span class=\\"fs-7\\">/&#x221E;</span></span><span class=\\"d-block fs-7\\">Volumes size</span></div></div></div></div></div>"`;
